### DISCUSSION TITLE
Windows could not start the DNSCrypt client proxy service on Local Computer.

### DISCUSSION THREAD

**Discussion by BenjaminMichaelis** [DNSCrypt/dnscrypt-proxy#2490] [Potential issues]
When starting the service, the following error occurs:

![image](https://github.com/DNSCrypt/dnscrypt-proxy/assets/22186029/70b634cf-153c-4205-9a02-1ea25a5ea82b)

Some logs from event viewer:
```
- The DNSCrypt client proxy service terminated unexpectedly.  It has done this 9 time(s).

- Check the remaining resource budget. Module exceeds resource budget, failed to AllocateFwCps, STATUS = Insufficient system resources exist to complete the API.
```

When starting from the command line with `.\dnscrypt-proxy -service start` it says ` [NOTICE] Service started`
But then if I send another command right after like `.\dnscrypt-proxy -service stop` or `.\dnscrypt-proxy -service restart` it says `[FATAL] Failed to restart DNSCrypt client proxy: The service has not been started.`.

Any suggestions?

**Comment by lifenjoiner** [comment]
There are bunch of reports on the net with the keyword `AllocateFwCps`. But, I don't see any clear reasons nor solutions.
Just unfortunately, it happens on your PC. Will you do some technical digging or searching around?

**Reply by BenjaminMichaelis** [reply]
Sure, is there anywhere particular I should look or anything in particular to do?

**Reply by lifenjoiner** [reply]
Possible ways may get more messages:
1. Turn on the logs.
2. Try CLI mode.
3. Debug it with a debugger.
4. See if there is any reports with the same environment (PC brand, OS and etc.) like yours.

**Reply by BenjaminMichaelis** [reply]
I think I got it working through those steps, thank you! I am not sure what changed but at some point it just started running. Is there a good way to test that traffic is actually moving through dnscrypt and windows isn't just using some dns server to resolve or such? Because it seems like even when the service is off, traffic somehow still resolves even after flushing the dns

**Reply by lifenjoiner** [reply]
Try the wiki: https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Checking

**Reply by BenjaminMichaelis** [reply]
> Try the wiki: https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Checking

I will try that, thanks!
